### PR TITLE
Techdebt: add ESLint rules to prepare for Vue 3 migration

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

* A part of https://github.com/nextcloud/spreed/issues/9448
* Also fix: https://github.com/nextcloud/spreed/issues/9449

Add ESLint rules for deprecated things from Vue 2.6 and using Vue 3 things available in Vue 2.7

Fix all errors.

### 🖼️ Screenshots

No changes

### CODE REVIEW

File: src/components/LeftSidebar/ConversationsList/Conversation.vue [hunk 1]
```diff
@@ -45,17 +45,14 @@
...
-				<Star v-if="item.isFavorite"
-					slot="icon"
-					:size="20" />
-				<Star v-else
-					slot="icon"
-					:size="20"
-					:fill-color="'#FFCC00'" />
+				<template #icon>
+					<Star v-if="item.isFavorite" :size="20" />
+					<Star v-else :size="20" :fill-color="'#FFCC00'" />
```
Comment (nickvergessen) [inline src/components/LeftSidebar/ConversationsList/Conversation.vue:54]:
```suggestion
					<Star v-else :size="20" fill-color="#FFCC00" />
```

😊

Comment (ShGKme) [reply src/components/LeftSidebar/ConversationsList/Conversation.vue:54]: I wanted to have this commit plain, having only one lint rule fix, related to the migration.

For this thing we can add a new rule with the auto fix.

**Comment by ShGKme** [comment]
Actually, it is ready to merge. Should we do it before the release?

**Comment by ShGKme** [comment]
Rebased onto master, added a fix for tests

**Review by Antreesy — APPROVED** [review]
Seems good by me!

File: src/components/LeftSidebar/NewGroupConversation/NewGroupConversation.vue [hunk 2]
```diff
@@ -21,8 +21,7 @@
 
 <template>
 	<div class="wrapper">
-		<NcButton slot="trigger"
-			v-tooltip.bottom="t('spreed', 'Create a new group conversation')"
```
Comment (ShGKme) [inline src/components/LeftSidebar/NewGroupConversation/NewGroupConversation.vue:25]: Note: `slot="trigger"` is just removed here. This button is not passed to any of the component's slots.

**Pull request merged**